Jl/add models word reference

jenniferlim opened the pull request.

lionel-alves commented on TranslateKit/Client.swift:
```diff
@@ -10,11 +10,14 @@ import Foundation
 public class Client {
 
     public let URLSession: NSURLSession
     public let baseURL = "http://api.urbandictionary.com/v0"
```
`urbanDictionaryBaseUrl`

lionel-alves commented on TranslateKit/Language.swift:
```diff
@@ -0,0 +1,32 @@
+            return "French"
+        case .English:
+            return "English"
+        }
```
Since `Language` has string raw values I think you can just do:

```
public func name() -> String {
    return self
}
```

jlimAP committed:
- Add model Translation, Meaning, Word, Language.
- Add new cassette Translate Word based on Word Reference reponse for Arm
- Add Translate function in Client
- Add test for Translate function

lionel-alves commented on TranslateKit/Meaning.swift:
```diff
@@ -0,0 +1,62 @@
+    public init?(dictionary: JSONDictionary) {
+        
+        guard let originalWordDictionary = dictionary["OriginalTerm"] as? JSONDictionary,
+        originalWord = Word(dictionary: originalWordDictionary) else { return nil }
```
indentation. :arrow_right:

lionel-alves commented on TranslateKit/Translation.swift:
```diff
@@ -0,0 +1,48 @@
+    
+    public init?(dictionary:JSONDictionary) {
+        guard let termDictionaries = dictionary["term0"] as? JSONDictionary,
+            meaningDictionaries = termDictionaries["PrincipalTranslations"] as? JSONDictionary else { return nil }
```
Newline.

lionel-alves commented on TranslateKit/Translation.swift:
```diff
@@ -0,0 +1,48 @@
+        self.meanings = sortedDictionaries.flatMap {
+            guard let dictionary = $0.1 as? JSONDictionary else { return nil }
+            return Meaning(dictionary: dictionary)
+        }
```
What about creating a method to do that since we do the exact same thing for `additionalMeaningDictionaries` and `compoundMeaningDictionaries`?

lionel-alves commented on TranslateKit/Word.swift:
```diff
@@ -0,0 +1,28 @@
+        self.sense = sense
+        self.usage = dictionary["usage"] as? String
+    }
+}
```
Newline.

lionel-alves commented on TranslateKitTests/TranslateKitTests.swift:
```diff
@@ -34,4 +35,90 @@ class TranslateKitTests: XCTestCase {
+        client.translate(word: "arm", from: .English, to: .French) { translation in
+            if let translation = translation {
+                
+                
```
:scissors:

lionel-alves commented on TranslateKitTests/TranslateKitTests.swift:
```diff
@@ -34,4 +35,90 @@ class TranslateKitTests: XCTestCase {
+            }
+            
+            expectation.fulfill()
+            }
```
indentation :point_left:

lionel-alves commented on TranslateKitTests/TranslateKitTests.swift:
```diff
@@ -34,4 +35,90 @@ class TranslateKitTests: XCTestCase {
+                // Principal Translation
+                XCTAssertEqual(translation.meanings.first?.translatedWords.first?.term, "bras")
+                XCTAssertEqual(translation.meanings.first?.translatedWords.first?.pos, "nm")
+                XCTAssertEqual(translation.meanings.first?.translatedWords.first?.sense, "corps")
```
Should we check the second translation too to check the order?

lionel-alves commented on TranslateKitTests/TranslateKitTests.swift:
```diff
@@ -34,4 +35,90 @@ class TranslateKitTests: XCTestCase {
+        
+        waitForExpectationsWithTimeout(1, handler: nil)
+    }
+}
```
Newline.

lionel-alves commented:
Left some comments, needs a rebase too.

jlimAP committed:
- Remove redundant code in Translation init function
- Clean up files by removing extra spaces, adding newline when appropriate

lionel-alves commented:
:bowtie:
